packages/fxa-settings/src/components/Modal/index.tsx

```diff
@@ -80,15 +84,24 @@ export const Modal = ({
                   Cancel
                 </button>
 
-                {onConfirm && (
-                  <button
-                    className={classNames('mx-2 flex-1', confirmBtnClassName)}
-                    data-testid="modal-confirm"
-                    onClick={onConfirm}
-                  >
-                    {confirmText}
-                  </button>
-                )}
+                {(onConfirm || route) &&
+                  (route ? (
+                    <Link
+                      className={classNames('mx-2 flex-1', confirmBtnClassName)}
+                      data-testid="modal-confirm"
+                      to={`${route}${location.search}`}
+                    >
+                      {confirmText}
+                    </Link>
+                  ) : (
+                    <button
+                      className={classNames('mx-2 flex-1', confirmBtnClassName)}
+                      data-testid="modal-confirm"
+                      onClick={onConfirm}
+                    >
+                      {confirmText}
+                    </button>
+                  ))}
```

Comment:
Wrapping everything in `(onConfirm || route)` might be redundant. This _could_ simplified to just:

```jsx
{route && (
  <Link
    className={classNames('mx-2 flex-1', confirmBtnClassName)}
    data-testid="modal-confirm"
    to={`${route}${location.search}`}
  >
    {confirmText}
  </Link>
)}

{onConfirm && (
  <button
    className={classNames('mx-2 flex-1', confirmBtnClassName)}
    data-testid="modal-confirm"
    onClick={onConfirm}
  >
    {confirmText}
  </button>
)}
```